Feature/main/home

Gzort opened the pull request.

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+import ru.itmo.edugoolda.core.widget.button.AppButton
+import ru.itmo.edugoolda.core.widget.button.ButtonType
+
+data class LessonListItem(
+    val name: String,
+    val time: String,
+    val result: String
+)
+
+data class PendingRequestsListItem(
+    val groupName: String,
+    val time: String
+)
```
Модельки должны лежать в `domain`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+)
+
+@Composable
+fun Home(
```
1) Это отдельная фича, должна лежать в своём модуле
2) UI экрана должен называться `HomeStudentUi`
3) У экрана должен быть свой компонент

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+
+@Composable
+fun Home(
+    lessonListItems: List<LessonListItem>,
+    requestListItems: List<PendingRequestsListItem>,
+    onAllResultsClick: () -> Unit,
+    onAllRequestsClick: () -> Unit,
```
В корневой `@Composable` экрана передаём `component`, а не данные и коллбеки

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+
+                )
+            Text(
+                text = "Ближайшие уроки",
```
Строки надо хранить в `strings.xml`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+            contentPadding = PaddingValues(12.dp)
+        ) {
+            Text(
+                text = "Все результаты",
```
Строки надо хранить в `strings.xml`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+
+                )
+            Text(
+                text = "Мои заявки",
```
Строки надо хранить в `strings.xml`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+            contentPadding = PaddingValues(12.dp)
+        ) {
+            Text(
+                text = "Все заявки",
```
Строки надо хранить в `strings.xml`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+            Text(
+                text = "EduGoolda",
+                style = CustomTheme.typography.caption.regular,
+                fontWeight = FontWeight.Bold,
+                fontSize = 52.sp
```
Это всё должно быть в `style`, либо выбирай подходящий, либо создавай новый

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+            Text(
+                text = "Ближайшие уроки",
+                style = CustomTheme.typography.caption.regular,
+                fontWeight = FontWeight.Bold,
+                fontSize = 16.sp
```
Это всё должно быть в `style`, либо выбирай подходящий, либо создавай новый

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Text(text = name, style = CustomTheme.typography.caption.regular, fontSize = 16.sp)
```
`fontSize` должен быть в `style`, либо выбирай подходящий, либо создавай новый

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+            Text(
+                text = result,
+                style = CustomTheme.typography.caption.regular,
+                fontWeight = FontWeight.Bold,
+                fontSize = 24.sp
```
Это всё должно быть в `style`, либо выбирай подходящий, либо создавай новый

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+            Text(
+                text = name,
+                style = CustomTheme.typography.caption.regular,
+                fontSize = 16.sp
```
`fontSize` должен быть в `style`, либо выбирай подходящий, либо создавай новый

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+                text = time,
+                style = CustomTheme.typography.caption.regular,
+                color = CustomTheme.colors.content.contentTertiary,
+                fontSize = 16.sp
```
`fontSize` должен быть в `style`, либо выбирай подходящий, либо создавай новый

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+            text = "На рассмотрении",
+            style = CustomTheme.typography.caption.regular,
+            color = CustomTheme.colors.content.contentTertiary,
+            fontSize = 16.sp
```
`fontSize` должен быть в `style`, либо выбирай подходящий, либо создавай новый

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+        }
+
+        Text(
+            text = "На рассмотрении",
```
Строки надо хранить в `strings.xml`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+import ru.itmo.edugoolda.core.widget.button.AppButton
+import ru.itmo.edugoolda.core.widget.button.ButtonType
+
+data class SolutionsListItem(
+    val owner: String,
+    val time: String,
+    val result: String
+)
+
+data class JoinRequestsListItem(
+    val owner: String,
+    val time: String,
+
+    )
```
Модельки должны лежать в domain

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+    )
+
+@Composable
+fun Home(
+    solutionItems: List<SolutionsListItem>,
+    requestItems: List<JoinRequestsListItem>,
+    onAllResultsClick: () -> Unit,
+    onAllRequestsClick: () -> Unit,
```
В корневой `@Composable` экрана передаём `component`, а не данные и коллбеки

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+    )
+
+@Composable
+fun Home(
```
1) Это отдельная фича, должна лежать в своём модуле
2) UI экрана должен называться `HomeStudentUi`
3) У экрана должен быть свой компонент

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Text(
+                text = "EduGoolda",
```
Строки надо хранить в `strings.xml`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+            Text(
+                text = "EduGoolda",
+                style = CustomTheme.typography.caption.regular,
+                fontWeight = FontWeight.Bold,
+                fontSize = 52.sp
```
Это всё должно быть в `style`, либо выбирай подходящий, либо создавай новый

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+                contentDescription = "View"
+            )
+            Text(
+                text = "Результаты выполнения",
```
Строки надо хранить в `strings.xml`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+                text = "Результаты выполнения",
+                style = CustomTheme.typography.caption.regular,
+                fontWeight = FontWeight.Bold,
+                fontSize = 16.sp
```
Это всё должно быть в `style`, либо выбирай подходящий, либо создавай новый

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+                .fillMaxWidth(),
+
+            buttonType = ButtonType.Secondary,
+            onClick = {},
```
Эта кнопка вроде должна что-то делать, лучше таки передавать в неё что-то, а в реализации коллбека поставить `// TODO`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+        // All Solutions button
+        AppButton(
+            modifier = Modifier
+                .clickable(onClick = onAllResultsClick)
```
Зачем ты повесил `.clickable()` на кнопку? У неё буквально параметр `onClick` есть

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+            contentPadding = PaddingValues(12.dp)
+        ) {
+            Text(
+                text = "Все результаты",
```
Строки надо хранить в `strings.xml`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+                contentDescription = "View"
+            )
+            Text(
+                text = "Заявки на вступление в группу",
```
Строки надо хранить в `strings.xml`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+                text = "Заявки на вступление в группу",
+                style = CustomTheme.typography.caption.regular,
+                fontWeight = FontWeight.Bold,
+                fontSize = 16.sp
```
Это всё должно быть в `style`, либо выбирай подходящий, либо создавай новый

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+        // All requests button
+        AppButton(
+            modifier = Modifier
+                .clickable(onClick = onAllRequestsClick)
```
Зачем ты повесил `.clickable()` на кнопку? У неё буквально параметр `onClick` есть

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+            contentPadding = PaddingValues(12.dp)
+        ) {
+            Text(
+                text = "Все заявки",
```
Строки надо хранить в `strings.xml`

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,294 @@
+            contentDescription = "View",
+        )
+    }
+}
```
Эти `@Composable` используются на экране учителя, и на экране студента, их надо переиспользовать + см. комменты, которые оставил на экране студента к ним

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+        // All Solutions button
+        AppButton(
+            modifier = Modifier
+                .clickable(onClick = onAllResultsClick)
```
Зачем ты повесил `.clickable()` на кнопку? У неё буквально параметр `onClick` есть

Krakozaybr commented on features/auth/src/main/java/ru/itmo/edugoolda/features/auth/home/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,289 @@
+        // All requests button
+        AppButton(
+            modifier = Modifier
+                .clickable(onClick = onAllRequestsClick)
```
Зачем ты повесил `.clickable()` на кнопку? У неё буквально параметр `onClick` есть

Gzort committed:
Create Home screen UI with component for teacher and student
Add res in home module
Delete home package in auth module
Create solution module in data

Gzort committed:
Delete useless comment

Gzort committed:
Delete HomeUI duplicate

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/student/HomeStudentComponentImpl.kt:
```diff
@@ -0,0 +1,57 @@
+
+    private val joinRequestsReplica = joinRequestRepository.joinRequestListReplica.toReplica()
+    private val solutionsReplica = solutionRepository.solutionListReplica.toReplica()
+    private val mainStateReplica = combine(
+        joinRequestsReplica,
+        solutionsReplica
+    ) { joinRequests, solutions -> listOf(joinRequests, solutions) }
```
`listOf` я по приколу тебе показывал, тут так-то совершенно разные типы. Создай
```kotlin
data class HomeStudentViewData(
    val joinRequests: ...,
    val solutions: ...
)
```

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/student/HomeStudentComponentImpl.kt:
```diff
@@ -0,0 +1,57 @@
+    ) { joinRequests, solutions -> listOf(joinRequests, solutions) }
+
+    override val mainState = mainStateReplica.observe(this, errorHandler)
+    override val joinRequestState =
+        joinRequestRepository.joinRequestListReplica.observe(this, errorHandler)
+    override val solutionState = solutionRepository.solutionListReplica.observe(this, errorHandler)
```
А вот эти убери, раз `combine` используешь

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/student/HomeStudentComponentImpl.kt:
```diff
@@ -0,0 +1,57 @@
+
+
+    override fun onSolutionClick(solution: Solution) {
+        TODO("Not yet implemented")
```
Добавь в `Communication` соответствующий метод

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/student/HomeStudentComponentImpl.kt:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    override fun onAllSolutionsClick() {
+        TODO("Not yet implemented")
```
Добавь в `Communication` соответствующий метод

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/student/HomeStudentComponentImpl.kt:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    override fun onAllJoinRequestsClick() {
+        TODO("Not yet implemented")
```
Добавь в `Communication` соответствующий метод

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/student/HomeStudentComponentImpl.kt:
```diff
@@ -0,0 +1,57 @@
+        mainStateReplica.revalidate()
+    }
+
+    override fun onLoadNext() {
+
+    }
```
У тебя негде пагинации быть не должно, убирай метод

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/student/HomeStudentUI.kt:
```diff
@@ -0,0 +1,196 @@
+            lazyListState.TriggerLoadNext(
+                pagedState = joinRequestsState,
+                hasNextPage = joinRequestsState.data?.hasNextPage == true,
+                callback = component::onLoadNext
+            )
+            LazyColumn(
+                state = lazyListState,
+
+                ) {
+                items(data.joinRequestList.take(3)) {
+                    JoinRequestStudentListItem(
+                        groupName = it.groupName,
+                        date = it.date
+                    )
+                }
+
+            }
+        }
+
+        // All requests button
+        AppButton(
+            modifier = Modifier
+                .fillMaxWidth(),
+
+            buttonType = ButtonType.Secondary,
+            onClick = { component.onAllJoinRequestsClick() },
+            contentPadding = PaddingValues(12.dp)
+        ) {
+            Text(
+                text = stringResource(id = homeR.string.all_join_requests),
+                style = CustomTheme.typography.button.bold,
+            )
+            Icon(
+                modifier = Modifier.padding(start = 12.dp),
+                painter = painterResource(R.drawable.ic_24_app_button_path),
+                contentDescription = null
+            )
+        }
+    }
+}
```
Раз решили делать единый стейт, нужен один `PullToRefresh`, который слушает как раз таки этот стейт `HomeStudentComponentImpl.mainState `

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/teacher/HomeTeacherComponent.kt:
```diff
@@ -0,0 +1,25 @@
+
+
+interface HomeTeacherComponent {
+    val mainState: StateFlow<LoadableState<List<Any>>>
+    val joinRequestState: StateFlow<PagedState<JoinRequestList>>
+    val solutionState: StateFlow<PagedState<SolutionList>>
```
Такой же коммент про единый стейт

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/teacher/HomeTeacherComponentImpl.kt:
```diff
@@ -0,0 +1,71 @@
+        }
+    }
+
+    override fun onSolutionClick(solution: Solution) {
+        TODO("Not yet implemented")
+    }
+
+    override fun onAllSolutionsClick() {
+        TODO("Not yet implemented")
+    }
+
+    override fun onAllJoinRequestsClick() {
+        TODO("Not yet implemented")
+    }
```
См. выше про `Communication`-методы

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/teacher/HomeTeacherComponentImpl.kt:
```diff
@@ -0,0 +1,71 @@
+        mainStateReplica.revalidate()
+    }
+
+    override fun onLoadNext() {
+
+    }
```
См. выше - удаляй

Krakozaybr commented on features/home/src/main/java/ru/itmo/edugoolda/features/home/presentation/teacher/HomeTeacherUI.kt:
```diff
@@ -0,0 +1,199 @@
+                callback = component::onLoadNext
+            )
+            LazyColumn(
+                state = lazyListState,
+
+                ) {
+                items(data.joinRequestList.take(3)) {
+                    JoinRequestTeacherListItem(
+                        groupName = it.groupName,
+                        studentName = it.sender.name,
+                        date = it.date,
+                        onAcceptClick = { component.onAcceptClick(it) },
+                        onDeclineClick = { component.onDeclineClick(it) }
+                    )
+                }
+
+            }
+        }
+
+        // All requests button
+        AppButton(
+            modifier = Modifier
+                .fillMaxWidth(),
+
+            buttonType = ButtonType.Secondary,
+            onClick = { component.onAllJoinRequestsClick() },
+            contentPadding = PaddingValues(12.dp)
+        ) {
+            Text(
+                text = stringResource(id = homeR.string.all_join_requests),
+                style = CustomTheme.typography.button.bold,
+            )
+            Icon(
+                modifier = Modifier.padding(start = 12.dp),
+                painter = painterResource(R.drawable.ic_24_app_button_path),
+                contentDescription = null
+            )
+        }
+    }
+}
```
См. выше коммент про единый стейт и PullToRefresh

Krakozaybr commented on features/join-requests/src/main/java/ru/itmo/edugoolda/features/invitations/presentation/JoinRequestsComponentImpl.kt:
```diff
@@ -18,6 +22,12 @@ class JoinRequestsComponentImpl(
 
     private val invitationReplica = joinRequestRepository.joinRequestListReplica
     override val joinRequestState = invitationReplica.observe(this, errorHandler)
+    val a = null
+    val mainJoinRequestsReplica = invitationReplica.toReplica().map { v -> v.joinRequestList.take(3)}
+    val mainStateReplica = combine(
+        mainJoinRequestsReplica,
+        invitationReplica.toReplica()
+    ) {mainJoinRequests, inv -> listOf(mainJoinRequests, inv) }
```
Это лишнее, полагаю

Gzort committed:
Add Communication in HomeComponent
Format code according detekt
Add HomeViewData

Gzort committed:
Change Column to LazyColumn

Seztor committed:
Merge branch 'develop' into feature/main/home

# Conflicts:
#	data/join_requests/src/main/java/ru/itmo/edugoolda/data/join_requests/internal/JoinRequestApi.kt
#	features/join_requests/src/main/java/ru/itmo/edugoolda/features/join_requests/presentation/JoinRequestsUI.kt
#	settings.gradle.kts

Seztor committed:
change JoinRequestTeacherListItem widget package
